actions/contacts/upsert.go

```diff
@@ -0,0 +1,60 @@
+package contacts
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/mappings"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoin-sv/spv-wallet/server/auth"
+	"github.com/gin-gonic/gin"
+)
+
+// upsert will add a new contact or modify an existing one.
+// Upsert contact godoc
+// @Summary		Upsert contact
+// @Description	Add or update contact. For new contact send request to add current user as contact
+// @Tags		Contact
+// @Produce		json
+// @Param		UpsertContact body contacts.UpsertContact true "Full name and metadata needed to add/modify contact"
+// @Success		201
+// @Router		/v1/contact/{paymail} [PUT]
+// @Security	x-auth-xpub
+func (a *Action) upsert(c *gin.Context) {
+	requesterPubKey := c.GetString(auth.ParamXPubKey)
+	cPaymail := c.GetString("paymail")
```

Comment:
Imo it should be:
cPaymail := c.Param("paymail")

Reply:
Fixed